static/app/views/performance/transactionSummary/transactionAnomalies/content.tsx

```diff
@@ -0,0 +1,266 @@
+import {WidgetEmptyStateWarning} from '../../landing/widgets/components/selectableList';
+import {QueryDefinition, WidgetDataResult} from '../../landing/widgets/types';
+import {
+  PerformanceWidgetSetting,
+  WIDGET_DEFINITIONS,
+} from '../../landing/widgets/widgetDefinitions';
+import {SetStateAction} from '../types';
+
+import AnomaliesTable from './anomaliesTable';
+import {AnomalyChart} from './anomalyChart';
+
+type Props = {
+  location: Location;
+  organization: Organization;
+  eventView: EventView;
+  projectId: string;
+  setError: SetStateAction<string | undefined>;
+  transactionName: string;
+};
+
+const anomalyAreaName = (anomaly: AnomalyInfo) => `#${anomaly.id}`;
+const transformAnomalyToArea = (
+  anomaly: AnomalyInfo
+): [{name: string; xAxis: number}, {xAxis: number}] => [
+  {name: anomalyAreaName(anomaly), xAxis: anomaly.start},
+  {xAxis: anomaly.end},
+];
+
+const transformAnomalyData = (_: any, results: {data: AnomalyPayload}) => {
+  const data: LineChartSeries[] = [];
+  const resultData = results.data;
+
+  if (!resultData) {
+    return {
+      isLoading: false,
+      isErrored: false,
+      data: undefined,
+      hasData: false,
+      loading: false,
+    };
+  }
+
+  data.push({
+    seriesName: 'tpm()',
+    data: resultData.y.data.map(([name, [{count}]]) => ({
+      name,
+      value: count,
+    })),
+  });
+  data.push({
+    seriesName: 'tpm() lower bound',
+    data: resultData.yhat_lower.data.map(([name, [{count}]]) => ({
+      name,
+      value: count,
+    })),
+  });
+  data.push({
+    seriesName: 'tpm() upper bound',
+    data: resultData.yhat_upper.data.map(([name, [{count}]]) => ({
+      name,
+      value: count,
+    })),
+  });
+
+  const anomalies = results.data.anomalies;
+  const highConfidenceAreas = anomalies
+    .filter(a => a.confidence === 'high')
+    .map(transformAnomalyToArea);
+  const highConfidenceLines = anomalies
+    .filter(a => a.confidence === 'high')
+    .map(area => ({xAxis: area.start, name: anomalyAreaName(area)}));
+
+  const lowConfidenceAreas = anomalies
+    .filter(a => a.confidence === 'low')
+    .map(transformAnomalyToArea);
+  const lowConfidenceLines = anomalies
+    .filter(a => a.confidence === 'low')
+    .map(area => ({xAxis: area.start, name: anomalyAreaName(area)}));
+
+  data.push({
+    seriesName: 'High Confidence',
+    color: theme.red300,
+    data: [],
+    silent: true,
+    markLine: MarkLine({
+      animation: false,
+      lineStyle: {color: theme.red300, type: 'solid', width: 1, opacity: 1.0},
+      data: highConfidenceLines,
+      label: {
+        show: true,
+        rotate: 90,
+        color: theme.red300,
+        position: 'insideEndBottom',
+        fontSize: '10',
+        offset: [5, 5],
+        formatter: obj => `${(obj.data as any).name}`,
+      },
+    }),
+    markArea: MarkArea({
+      itemStyle: {
+        color: color(theme.red300).alpha(0.2).rgb().string(),
```

Comment:
Should this use the builtin [opacity](https://echarts.apache.org/en/option.html#series-line.markArea.itemStyle.opacity) instead?

Reply:
I'll check 👍